url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/bitfield/TestForBitField.java

```diff
@@ -0,0 +1,371 @@
+		{
+			BitField field = newBitField(0);
+			assertEquals(-1, field.getOnPieceAtRandom());
+		}
+		{
+			BitField field = newBitField(1);
+			assertEquals(0, field.getOnPieceAtRandom());
+			field.isOn(0, false);
+			assertEquals(-1, field.getOnPieceAtRandom());
+		}
+
+		{
+			BitField field = newBitField(3);
+			int i = field.getOnPieceAtRandom();
+			assertEquals(true, (i==0|i==1||i==2?true:false));
+
+			field.isOn(1, false);
+			 i = field.getOnPieceAtRandom();
+			assertEquals(true, (i==0||i==2?true:false));
+
+			field.isOn(2, false);
+			 i = field.getOnPieceAtRandom();
+			assertEquals(i, 0);
+			
+			field.isOn(0, false);
+			 i = field.getOnPieceAtRandom();
+			assertEquals(-1, i);
+		}
+	}
+
+	public void test_relative() {
+		{
+			BitField myinfo = new BitField(22);
+			BitField target = new BitField(22);
+			BitField output = new BitField(22);
+			myinfo.oneClear();
+			target.oneClear();
+			BitField.relative(target, myinfo, output);
+			for(int i=0;i<22;i++) {
+				assertEquals(false, output.isOn(i));
+			}
+		}
+		{
+			BitField myinfo = new BitField(22);
+			BitField target = new BitField(22);
+			BitField output = new BitField(22);
+			myinfo.zeroClear();
+			target.zeroClear();
+			BitField.relative(target, myinfo, output);
+			for(int i=0;i<22;i++) {
+				assertEquals(false, output.isOn(i));
+			}
+		}
+
+		{
+			BitField myinfo = new BitField(22);
+			BitField target = new BitField(22);
+			BitField output = new BitField(22);
+			myinfo.zeroClear();
+			target.oneClear();
+			BitField.relative(target, myinfo, output);
+			for(int i=0;i<22;i++) {
+				assertEquals("["+i+"]", true, output.isOn(i));
+			}
+		}
+		{
+			BitField myinfo = new BitField(22);
+			BitField target = new BitField(22);
+			BitField output = new BitField(22);
+			myinfo.oneClear();
+			target.zeroClear();
+			BitField.relative(target, myinfo, output);
+			for(int i=0;i<22;i++) {
+				assertEquals("["+i+"]", false, output.isOn(i));
+			}
+		}
+
+		{
+			BitField myinfo = new BitField(22);
+			BitField target = new BitField(22);
+			BitField output = new BitField(22);
+			myinfo.zeroClear();
+			target.zeroClear();
+			target.isOn(1, true);
+			BitField.relative(target, myinfo, output);
+			
+			assertEquals("["+0+"]", false, output.isOn(0));
+			assertEquals("["+1+"]", true, output.isOn(1));
+			for(int i=2;i<22;i++) {
+				assertEquals("["+i+"]", false, output.isOn(i));
+			}
+		}
+	}
+
+	public void testUnique() {
+		{
+			BitField target = new BitField(1974);
+			target.zeroClear();
+			int[] indexs={
+				434, 1157, 1455, 74, 764, 1414, 1941, 1723, 
+				446, 1442, 955, 408, 1577, 377, 767, 223,
+				1328, 629, 1875, 203, 1244, 1847, 753, 1332
+			};
+			for(int index: indexs) {
+				target.isOn(index, true);
+			}
+			A:for(int j=0;j<100;j++) { 
+				int ret = target.getOnPieceAtRandom();
+				for(int index:indexs) {
+					if(index==ret) {
+						break A;
+					}
+				}
+				assertTrue("[["+ret+"]]", false);
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 259:267, 291:299.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.